addons/account/models/ir_sequence.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class IrSequence(models.Model):
+    _inherit = 'ir.sequence'
+
+    @api.constrains('prefix', 'suffix')
+    def _validate_sequence_number(self):
+        journal_ids = self.env['account.journal'].search([('restrict_mode_hash_table', '=', True)])
+        for seq in self:
+            if any(seq.id == journal.secure_sequence_id.id and (not seq.next_by_id().strip().isdigit()) for journal in journal_ids):
```

Comment:
* using `next_by_id` will actually consume the next identifier, we should just check if prefix and suffix are not digits 
* looping on journals is not really necessary. If in the search we add `('secure_sequence_id', 'in', self.ids)` then we could just check the retrieved journal_ids